drltrace/CMakeLists.txt

```diff
@@ -31,48 +31,86 @@
 
 cmake_minimum_required(VERSION 2.6)
 
-# We do not need libc, and we avoid a 10x size increase in both our
-# dll and pdb (plus we avoid stressing private library isolation) by
-# not using it (i#714).
-set(DynamoRIO_USE_LIBC OFF)
+include(${PROJECT_SOURCE_DIR}/make/policies.cmake NO_POLICY_SCOPE)
 
-add_library(drltrace SHARED drltrace.c)
-configure_DynamoRIO_client(drltrace)
-use_DynamoRIO_extension(drltrace drmgr)
-use_DynamoRIO_extension(drltrace drwrap)
-use_DynamoRIO_extension(drltrace drx)
-# We keep our shared libs in the lib dir, not the bin dir:
-place_shared_lib_in_lib_dir(drltrace)
+##################################################
+# drltracelib
 
+set(srcs
+    drltrace.c)
+
+set(external_srcs ../framework/drmf_utils.c)
+
+if (WIN32)
+  set(srcs ${srcs} ${PROJECT_SOURCE_DIR}/make/resources.rc)
+  set(DEFINES_NO_D ${DEFINES_NO_D} RC_IS_DRLTRACE)
+  # We set DR libc option on b/c this is needed to link with drsyms_static
+  # on VS2010. It has a downside of increase in pdb and dll size (xref DRi#714).
+  set(DynamoRIO_USE_LIBC ON)
+endif ()
+
+add_library(drltracelib SHARED ${srcs} ${external_srcs})
+
+# We share the framework version # for now
+set_library_version(drltracelib ${DRMF_VERSION_MAJOR_MINOR})
+
+if (WIN32)
+  set_property(TARGET drltracelib PROPERTY COMPILE_DEFINITIONS
+    "${DEFINES_NO_D};RC_IS_DRLTRACELIB")
+else ()
+  set_property(TARGET drltracelib PROPERTY COMPILE_DEFINITIONS ${DEFINES_NO_D})
+endif ()
+
+set_target_properties(drltracelib PROPERTIES
+  RUNTIME_OUTPUT_DIRECTORY${location_suffix} "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+
```

Comment:
Sorry, I forgot set(DynamoRIO_RPATH ON) here. Otherwise drltrace doesn't know know how to find libdynamorio.so.